src/main/java/com/ethan/twfaith/powers/blessings/HellsFury.java

```diff
@@ -20,15 +20,16 @@ public void onHellsFuryTrigger(Player player, PlayerData player_data, ItemStack
         for (Player nearby : Bukkit.getOnlinePlayers()){
             PlayerData nearby_data = PlayerHashMap.playerDataHashMap.get(nearby.getUniqueId());
             if (!player.getWorld().equals(nearby.getWorld())){continue;}
-            if (player.getLocation().distance(nearby.getLocation()) >= 30 || !player_data.getLedBy().equals(nearby_data.getLedBy())){continue;}
-            if (nearby_data.isHellsFuryActive()){
+            if (nearby_data.isHellsFuryActive() && player_data.getLedBy().equals(nearby_data.getLedBy())){
+                continue;
                 nearby_data.setHellsFuryActive(false);
                 player.sendMessage("Your foot flames cease.");
                 if (!chosen_item.getEnchantments().isEmpty()){
                     chosen_item_meta.removeEnchant(Enchantment.DURABILITY);
                     chosen_item.setItemMeta(chosen_item_meta);
                 }
-            }else{
+            }else if (player.getLocation().distance(nearby.getLocation()) >= 30 || !player_data.getLedBy().equals(nearby_data.getLedBy())){
```

Comment:
Good job moving the distance check to the appropriate block. However, this check was originally intended to filter out players who were not supposed to get the power. Because you didn't invert this check, it would actually activate the power for only the wrong people on the server.